src/views/AddItem.jsx

```diff
@@ -29,19 +32,46 @@ export function AddItem({ listToken }) {
 
 	// Store values inside itemData and sent over to database
 	// otherwise log an error if request fails
-	const onFormSubmit = (event) => {
-		event.preventDefault();
+	const onFormSubmit = (e) => {
+		e.preventDefault();
 
 		// if list token is not provided, notify user and exit function
 		if (!listToken) {
 			notifyNoToken();
 			return;
 		}
 
+		// `itemRegex` is a regular expression that matches any sequence of characters
+		// that does NOT include letters or digits
+		const itemRegex = /[^a-zA-Z0-9]+/g;
+
+		// We are looking for an array of items that matches the user input.
+		// We use `toLowerCase`, `replaceAll` and `itemRegex` to account for
+		// all the edge cases: white space, letter case, and non-letters/non-digits.
+		const nameMatchesArr = data.some((item) => {
+			return (
+				item.name &&
+				item.name.toLowerCase().replaceAll(itemRegex, '') ===
+					itemName.toLowerCase().replaceAll(itemRegex, '')
+			);
+		});
```

Comment:
I'm on the fence about this, but would it make sense to also replace or account for adding numbers to items? I think it's beyond the scope of this issue, but it may be worth considering. I was able to add flour7 to a list that already had flour. It may cause other issues if there are cases where similar items may only be differentiated by a number, but I'm also curious on others' thoughts.

Reply:
@sarahmjenkins Thanks Sarah! I was also on the fence about that one. You’re absolutely right, I was taking into account names that might involve numbers. Because what if, for some reason, there WAS a brand called Flour7? Lol. Mentors, if you’re reading this, what do you think?

Reply:
in my humble opinion, this is not a case you need to be concerned about 🤷